File: app/src/androidTest/java/in/mvpstarter/sample/util/ErrorTestUtil.java
package in.mvpstarter.sample.util;

import in.mvpstarter.sample.R;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static org.hamcrest.Matchers.allOf;

public class ErrorTestUtil {

    public static void checkErrorViewsDisplay() {
        onView(allOf(withText(R.string.error_title), isDisplayed()))
                .check(matches(isDisplayed()));
        onView(allOf(withText(R.string.error_message), isDisplayed()))
                .check(matches(isDisplayed()));
    }

    public static void checkClickingReloadShowsContentWithText(String expectedText) {
        onView(allOf(withText(R.string.error_reload), isDisplayed()))
                .perform(click());
        onView(withText(expectedText))
                .check(matches(isDisplayed()));
    }

}

File: app/src/androidTest/kotlin/in/mvpstarter/sample/util/ErrorTestUtil.kt
package `in`.mvpstarter.sample.util

import `in`.mvpstarter.sample.R

import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.click
import android.support.test.espresso.assertion.ViewAssertions.matches
import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
import android.support.test.espresso.matcher.ViewMatchers.withText
import org.hamcrest.Matchers.allOf

object ErrorTestUtil {

    fun checkErrorViewsDisplay() {
        onView(allOf<View>(withText(R.string.error_title), isDisplayed()))
                .check(matches(isDisplayed()))
        onView(allOf<View>(withText(R.string.error_message), isDisplayed()))
                .check(matches(isDisplayed()))
    }

    fun checkClickingReloadShowsContentWithText(expectedText: String) {
        onView(allOf<View>(withText(R.string.error_reload), isDisplayed()))
                .perform(click())
        onView(withText(expectedText))
                .check(matches(isDisplayed()))
    }

}
